pm: device_runtime: put_sync_locked: fix usage count on err

bjarki-andreasen opened the pull request:
If pm action fails within put_sync_locked, usage count should be reset to reflect the state of the device.

bjarki-andreasen-nordic committed:
tests: pm: device_runtime_api: add retval to test_driver

Add API to configure the return value of pm actions of the
test_driver used by the device_runtime_api test suite.

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

bjarki-andreasen-nordic committed:
pm: device_runtime: put_sync_locked: fix usage count on err

If pm action fails within put_sync_locked, usage count should be
reset to reflect the state of the device.

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

JordanYates reviewed (changes requested):
This should have a testcase covering it

bjarki-andreasen-nordic committed:
tests: pm: device_runtime_api: test neg driver retval

Test the test driver returning negative values on calls to
pm_device_runtime_get/put calls.

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>
